lib/Runtime/PlatformAgnostic/Platform/Linux/ArraysUtility.cpp

```diff
@@ -0,0 +1,81 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Common.h"
+#include "ChakraPlatform.h"
+#include <locale.h>
+
+namespace PlatformAgnostic
+{
+namespace Arrays
+{
+    #define BufSize 256
+
+    template<class T, class N = T>
+    void ByteCopy(const T *from, N *to, const size_t max_length)
+    {
+        size_t length = 0;
+        for (; length < max_length && (to[length] = static_cast<N>(from[length])) != 0;
+            length++);
+        if (length == max_length)
+        {
+            to[length - 1] = N(0);
+        }
+    }
+
+    Utility::ArraysLocale::ArraysLocale()
+    {
+
+        commaSeparator = WCHAR(',');
+        semicolonSeparator = WCHAR(';');
+
+        char buffer[BufSize];
+        char *old_locale = setlocale(LC_NUMERIC, NULL);
+
+        if (old_locale != NULL)
+        {
+            ByteCopy<char>(old_locale, buffer, BufSize);
```

Comment:
Any reason for not just using `memcpy`?

Reply:
No reason actually. I was just using PlatformAgnostic::Numbers as a reference when I wrote it.

Reply:
Oh, that's... weird. I can't imagine why Numbers wouldn't also use memcpy. I don't think you need to change Numbers, but I don't think that's a pattern that should be encouraged :P